### DISCUSSION TITLE
Where should layout visualizations live?

### DISCUSSION THREAD

**Discussion by paulf81** [NatLabRockies/floris#789] [v4 Design Discussion]
Currently there are layout visualization codes in both FLORIS and FLASC, with the FLASC versions having more capability over the FLORIS version.  @misi9170 and I were discussing how we might manage this and we had a few options in mind:

1) FLORIS has simple layout, FLASC imports this and augments it
2) All visualization of layout moved to FLORIS.  Our reasoning here is FLASC requires FLORIS, but FLORIS doesn't require FLASC so if we can envision a FLORIS use case for better layout visualizations seperate from SCADA analysis, it could belong in FLORIS
3) A new separate visualization repo that both import

Tagging a few folks to get your comments: @misi9170 @ejsimley @rafmudaf @bayc @christiannvaughn @Bartdoekemeijer

**Comment by rafmudaf** [comment]
I vote to strike option 3 on the grounds of simplicity. Moving the visualization tools out of both means more work to keep them all in sync (mostly releases and when either FLORIS or FLASC change an API now you have to go change another repo).
As for maintaining the status quo (option 1) or consolidating all to FLORIS (option 2), I'll have to think on it a bit more.

**Reply by paulf81** [reply]
I think I favor pushing all into FLORIS.  I can imagine FLORIS use cases where the direction lines between turbines is useful that are not connected to SCADA and so having it in FLORIS is the most versatile option.  But open to debate

**Comment by Bartdoekemeijer** [comment]
My vote is **option 2**. FLASC is really meant for SCADA analysis. The visualization functions in FLASC are not limited to that purpose, but really are generic functions that can be used for any FLORIS application.

**Comment by ejsimley** [comment]
I like Option 2 as well. It seems like the layout and power and thrust curve plotting functions in `visualization` could be moved to FLORIS. Some visualization capabilities like those in `yaw_optimizer_visualization` are more analysis specific and should probably stay in FLASC.

**Comment by paulf81** [comment]
Seems like we have a good consensus!  @rafmudaf I have a technical question for you, this is a change which effects two repos (floris and flasc), is the right way to track this change seperate issues in each repo which link to each other?

**Comment by misi9170** [comment]
#805 moved layout visualizations from FLASC to FLORIS
